R/carehomes.R

```diff
@@ -254,26 +259,44 @@ carehomes_compare <- function(state, prev_state, observed, pars) {
   model_sympt_cases <- state["sympt_cases", ] - prev_state["sympt_cases", ]
+  model_sympt_cases_over25 <- state["sympt_cases_over25", ] -
+    prev_state["sympt_cases_over25", ]
   model_react_pos <- state["react_pos", ]
 
-  ## Add some small exponential noise in case the number of model cases is 0
-  pillar2_pos <- model_sympt_cases +
-    rexp(length(model_sympt_cases), pars$observation$exp_noise)
-  pillar2_neg <- pars$prop_noncovid_sympt *
-    (sum(pars$N_tot) - model_sympt_cases)
-  model_pillar2_prob_pos <- (pars$pillar2_sensitivity * pillar2_pos +
-                               (1 - pars$pillar2_specificity) * pillar2_neg) /
-    (pillar2_pos + pillar2_neg)
-
-  ## Note that for REACT we exclude group 1 (0-4) and 19 (CHR)
-  react_pos <- model_react_pos +
-    rexp(length(model_react_pos), pars$observation$exp_noise)
-  react_neg <- sum(pars$N_tot[2:18]) - model_react_pos
-  model_react_prob_pos <- (pars$react_sensitivity * react_pos +
-                             (1 - pars$react_specificity) * react_neg) /
-    (react_pos + react_neg)
+  ## calculate test positive probabilities for the various test data streams
+  ## Pillar 2
+  pillar2_negs <- pars$prop_noncovid_sympt * (sum(pars$N_tot)
+                                              - model_sympt_cases)
+  model_pillar2_prob_pos <- test_prob_pos(model_sympt_cases,
+                                          pillar2_negs,
+                                          pars$pillar2_sensitivity,
+                                          pars$pillar2_specificity,
+                                          pars$observation$exp_noise)
+
+  ## Pillar 2 over 25s
+  pillar2_over25_negs <- pars$prop_noncovid_sympt * (sum(pars$N_tot[6:19])
+                                              - model_sympt_cases_over25)
+  model_pillar2_over25_prob_pos <- test_prob_pos(model_sympt_cases_over25,
+                                                 pillar2_over25_negs,
+                                                 pars$pillar2_sensitivity,
+                                                 pars$pillar2_specificity,
+                                                 pars$observation$exp_noise)
+
+  ## REACT (Note that for REACT we exclude group 1 (0-4) and 19 (CHR))
+  model_react_prob_pos <- test_prob_pos(model_react_pos,
+                                        sum(pars$N_tot[2:18]) - model_react_pos,
+                                        pars$react_sensitivity,
+                                        pars$react_specificity,
+                                        pars$observation$exp_noise)
+
+  ## serology
+  model_sero_prob_pos <- test_prob_pos(model_sero_pos,
+                                       pars$N_tot_15_64 - model_sero_pos,
+                                       pars$sero_sensitivity,
+                                       pars$sero_specificity,
+                                       pars$observation$exp_noise)
 
   pars <- pars$observation
   exp_noise <- pars$exp_noise
```

Comment:
Could maybe move this up, say to line 267 so you can use the unpacked exp_noise object throughout the calls to test_prob_pos()

This is v. minor though!